lib/src/facts/linux/virtualization_resolver.cc

```diff
@@ -25,6 +21,31 @@ namespace lth_file = leatherman::file_util;
 
 namespace facter { namespace facts { namespace linux {
 
+    string virtualization_resolver::get_cloud_provider(collection& facts)
+    {
+        // Check for Azure
+        std::string provider = get_azure(facts);
+
+        return provider;
+    }
+
+    string virtualization_resolver::get_azure(collection& facts, string const& leases_file)
+    {
+        std::string provider;
+        if ( boost::filesystem::exists( leases_file ) )
+        {
+            lth_file::each_line(leases_file, [&](string& line) {
+                if (line.find("option 245") != std::string::npos ||
```

Comment:
https://social.msdn.microsoft.com/Forums/azure/en-US/f7fbbee6-370a-41c2-a384-d14ab2a0ac12/what-is-the-correct-method-in-linux-on-azure-to-test-if-you-are-an-azure-vm-?forum=WAVirtualMachinesforWindows is the source, plus a few other places.